### PR TITLE
feat(sandbox): make enforce posture non-droppable

### LINKED ISSUE #2778: sandbox: make enforce posture non-droppable
Part of the sandbox hardening umbrella (T0).

## Problem

The current unsandboxed state was reached silently. `agent.sandbox_mode: ""` is a valid config, `DefaultSandboxMode()` maps it to `report`, and `report` is indistinguishable from `enforce` in every user-visible way except the absence of a wrapper around the spawn. Nothing failed, nothing warned loudly enough to notice, and `CLAUDE.md` drifted into claiming a posture that was never active.

Flipping the config (T0a) fixes today. This issue makes the posture non-droppable so it cannot drift back.

## Scope

- **Fail-closed validation**: reject an *unstated* posture for unattended/server runs — the posture must be chosen, never inferred. `internal/config/validation.go` already validates `sandbox_mode` and asserts `enforce` requires darwin/linux (`validation.go:89-92`); add the converse — a headless/server instance refusing to start on `off`/`report` unless an explicit override is set. `report` should remain available as a deliberate, loud opt-out for local debugging, not as the value an empty field silently lands on.
- **Startup visibility**: include the resolved sandbox posture in the existing `app.automations` startup summary line, so the posture of each instance is verifiable at a glance the same way its automation roles are.
- **Escape-hatch accountability**: the per-task `sandbox: false` hatch currently takes a bare bool. Require a reason string, and carry it into `audit.EventAgentSandboxDisabled` alongside the existing `agentorch.logSandboxEscapeHatch` warning, so a disabled sandbox is explainable after the fact rather than merely detectable.

## Verification

- A server config with no `sandbox_mode` fails validation with a clear message. An explicit `report` or `off` is accepted: requiring `enforce` outright would refuse to boot every `deploy/k8s-poc/` config (all four legitimately run `off`) and any local debug server, and this issue's own scope calls for `report` to stay available as a deliberate opt-out.
- Startup log line states the resolved posture.
- Setting `sandbox: false` without a reason is rejected; with one, the reason appears in the audit event.

### PR DESCRIPTION
**Pull request by Automaat**
## Motivation

An unset `agent.sandbox_mode` resolves to `report`, which validates the write allowlist and logs it but never wraps the spawn. So a config that simply omits the key is indistinguishable from a contained one until you grep the logs — which is how the server ran unsandboxed for months while the docs claimed `enforce`. Closes #2778, part of #2775.

> Changelog: feat(sandbox): require an explicit sandbox posture on the server

## Implementation information

Three changes:

**The posture must be stated.** `config.RequireExplicitSandboxMode(true)`, set once by `sybra-server`, makes an unset `agent.sandbox_mode` a load error. Any of `off`/`report`/`enforce` is accepted — the requirement is that an operator chose it. Server-only, so the desktop app and CLI keep the `report` default.

The check sits in the config **load path**, not at the server's startup call site. A boot-only check is defeated by the fsnotify config watcher: `ReloadFromDisk` → `LoadNoPersist` → `ReplaceRuntimeConfig` hot-applies `agent.sandbox_mode`, so deleting the line at runtime would silently drop the fleet back to `report` with no restart and no gate. Putting it in `load()` covers startup, the `-check-config` deploy preflight (so `sybra-build.sh` quarantines a bad candidate instead of promoting one that then restart-loops), and every hot reload — where a rejected load already restores last-known-good.

It is deliberately *not* in `ValidateResolvedConfig`: that also resolves the built-in empty config, so folding it in there panicked every `DefaultConfig` caller. There is a regression test for exactly that.

**Startup states the posture.** `app.automations` now logs `sandbox_mode`, so an instance's containment is visible at a glance alongside its automation roles.

**The escape hatch says why.** `task.SandboxOffReason` is carried into `audit.EventAgentSandboxDisabled`. It has a writer on the supported path — `UpdateFromMap` rejects the whole map on an unknown key, so without a `sandbox_off_reason` case, an operator following the field's documentation would have had their `sandbox: false` flip rejected too, breaking the hatch.

Deliberately *not* hard-rejecting a reasonless `sandbox: false`: the hatch exists for emergencies and must not become unusable. An empty `"reason":""` in the audit line marks an unexplained bypass as plainly as a separate flag would.

## Testing

`mise run verify`. Both new gates were mutation-tested — deleting the `ValidateUnattendedPosture` call site fails the load-path test, and flipping the escape-hatch assertion failed before it was corrected.

## Open questions

`sybra-cli config doctor` still reports clean on an unset `sandbox_mode`. Left as-is: the deploy preflight now fails loudly on the same condition, which is a stronger signal than an advisory warning.

Two unrelated failures surfaced while running the gate, filed rather than fixed here: #2802 (`TestE2E_WaitHuman_InvalidActionRejected`, currently red on `main`) and #2804 (`diskreclaim` TempDir cleanup race). #2802 may make this PR's E2E job red for reasons unrelated to the diff.

### CODE REVIEW

File: internal/task/store.go [hunk 1]
```diff
@@ -550,6 +550,9 @@ func applyCreateInit(t *Task, init Update, now time.Time) {
 	if init.ForkSubagent != nil {
 		t.ForkSubagent = *init.ForkSubagent
 	}
+	if init.SandboxOffReason != nil {
+		t.SandboxOffReason = *init.SandboxOffReason
+	}
 	if init.Sandbox != nil {
 		t.Sandbox = init.Sandbox
 	}
```
[the remark this answers is not in the document]
Comment (Automaat) [reply internal/task/store.go:561]: Fixed. Both call sites now run through `normalizeSandboxEscapeHatch`, which drops the reason whenever the hatch is not actually off, so a task can no longer carry a justification for a sandbox that is enabled.

Worth naming the consequence you implied: because the reason is only kept while `sandbox` is false, the flip and its reason have to arrive in the same call. Setting `sandbox: false` in one request and the reason in a later one is now refused rather than silently accepted half-done. That seemed better than persisting an unexplained bypass and hoping the second call arrives.

File: internal/sybra/agentorch/agentorch.go [hunk 2]
```diff
@@ -493,9 +493,13 @@ func (o *Orchestrator) logSandboxEscapeHatch(taskID string, t task.Task) {
...
 	}
-	o.logger.Warn("agent.sandbox.escape_hatch", "task_id", taskID)
+	// An empty reason is itself the signal: `"reason":""` in the audit line
+	// marks an unexplained bypass as plainly as a separate flag would.
+	reason := strings.TrimSpace(t.SandboxOffReason)
+	o.logger.Warn("agent.sandbox.escape_hatch", "task_id", taskID, "reason", reason)
 	o.LogAudit(audit.EventAgentSandboxDisabled, taskID, "", map[string]any{
 		"configured_default": o.cfg.DefaultSandboxMode(),
+		"reason":             reason,
 	})
```
[the remark this answers is not in the document]
Comment (Automaat) [reply internal/sybra/agentorch/agentorch.go:503]: You and the pre-PR adversarial reviewer independently landed on this, so I've implemented the rejection rather than defending the softer behaviour.

`normalizeSandboxEscapeHatch` (internal/task/store.go) now refuses `sandbox: false` with a blank or whitespace-only reason, on both the create and update paths — the boundary an operator actually calls. My earlier argument for record-and-flag was that a hard reject would make the hatch unusable in an emergency because there is no GUI control for it; that argument was wrong, since there is no GUI control for `sandbox` either, so the hatch was already CLI/frontmatter-only and the reason can be required on the same call.

`logSandboxEscapeHatch` stays permissive deliberately: it is a dispatch-time observer, and a pre-existing or hand-edited task file must still be *reported* rather than silently dropped. Enforcement belongs at the mint boundary, matching the `ValidateMintableAgentMode` pattern already in the repo.

File: internal/config/validation.go [hunk 3]
```diff
@@ -36,6 +37,47 @@ func (e *ValidationError) Error() string {
...
+	if cfg == nil || !requireExplicitSandboxMode.Load() {
+		return nil
+	}
+	if strings.TrimSpace(cfg.Agent.SandboxMode) != "" {
+		return nil
+	}
+	return &ValidationError{Messages: []string{
+		"agent.sandbox_mode is unset; set it explicitly to one of off, report, enforce " +
+			"(enforce is the contained posture — off and report leave agent writes unrestricted)",
+	}}
```
[the remark this answers is not in the document]
Comment (Automaat) [reply internal/config/validation.go:78]: Deliberate, and the issue is self-contradictory here — I'll fix the issue rather than the code.

#2778's scope says both "reject a non-enforce posture for unattended/server runs" and, two sentences later, "\`report\` should remain available as a deliberate, loud opt-out for local debugging, not as the value an empty field silently lands on." Those cannot both hold. The second is the one worth keeping: requiring `enforce` outright would refuse to boot every k8s PoC config in `deploy/k8s-poc/` (all four legitimately run `sandbox_mode: "off"`) and any local debug server.

So the contract implemented is: the posture must be *stated*, never inferred. That closes the actual defect — an omitted key silently resolving to `report` and looking identical to a contained config. I'm updating the issue text and its verification criteria to match.
